client-app/shared/static-content/composables/useTemplates.ts

```diff
@@ -1,19 +1,18 @@
 import { createGlobalState, useMemoize } from "@vueuse/core";
-import { ref } from "vue";
 import { useFetch } from "@/core/api/common";
-import type { PageTemplate, PageTemplateRequest } from "@/shared/static-content/types";
+import type { PageTemplate } from "@/shared/static-content/types";
 
-function _useTemplates() {
-  const payload = ref<PageTemplateRequest>();
-
-  const { data, execute: loadTemplate } = useFetch("/storefrontapi/content/templates", { immediate: false })
-    .post(payload)
-    .json<PageTemplate>();
+//list of files names in client-app/public/content/templates
+const TEMPLATES = ["product"];
 
+function _useTemplates() {
   const memoize = useMemoize(async (templateName: string) => {
-    payload.value = { template: templateName };
-    await loadTemplate();
-    return data.value;
+    if (!TEMPLATES.includes(templateName as string)) {
+      throw new Error(`Template with name ${templateName}.json not found!`);
+    }
+
+    const { data } = await useFetch(`/content/templates/${templateName}.json`).get().json();
```

Comment:
This API is no longer working with storefront as you moved files to different folder and you do `GET` instead of `POST`